mlflow/pyfunc/__init__.py

```diff
@@ -1580,9 +1593,9 @@ def batch_predict_fn(pdf, params=None):
                 return client.invoke(pdf).get_predictions()
 
         elif env_manager == _EnvManager.LOCAL:
-            if is_spark_connect:
+            if is_spark_connect and not should_spark_connect_use_nfs:
                 model_path = os.path.join(
-                    tempfile.gettempdir(),
+                    create_tmp_dir(),
```

Comment:
Does this create a unique directory in each invocation?

Reply:
Good catch. For non-databricks environment, it returns different path each invocation.
I move this path generation to driver side.

Reply:
emm, moving to driver side might have other issue, driver side might has different OS with executor side, then legal tmp path in driver side might not be available in executor side. e.g. macOS returns temp path like `'/var/folders/z7/47xgtbq53kg8hs1zn7qk8kw40000gp/T/tmptdo5n4l5'`

Reply:
I will revert it to `tempfile.gettempdir()` for now.

Reply:
This branch code will be removed once `spark_session.add_artifacts` API is ready in databricks shared cluster. See https://databricks.atlassian.net/browse/ML-36703